src/types.ts

```diff
@@ -91,6 +92,11 @@ export type LastActiveToolBeforeEraser =
   | null;
 
 export type AppState = {
+  contextMenu: {
+    options: (ContextMenuOption | false | null | undefined)[];
```

Comment:
Don't know if the plan when rendering is to check if `this.state.contextMenu.options.length > 0`, but I'd make the type into `contextMenu: {/* ... */} | null` where `null` indicates the contextMenu is closed.